src/Components/Samples/BlazorServerApp/Pages/FetchData.razor

```diff
@@ -41,6 +44,14 @@ else
 
     protected override async Task OnInitializedAsync()
     {
-        forecasts = await ForecastService.GetForecastAsync(DateTime.Now);
+        if (!ApplicationState.TryRetrievePersistedState("fetchdata", out var data))
+        {
+            forecasts = await ForecastService.GetForecastAsync(DateTime.Now);
+            ApplicationState.PersistState("fetchdata", JsonSerializer.SerializeToUtf8Bytes(forecasts));
+        }
+        else
+        {
+            forecasts = JsonSerializer.Deserialize<WeatherForecast[]>(data);
+        }
```

Comment:
This is a sample using "the most basic" APIs to preserve the state in any way you want.